source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -331,14 +417,30 @@ void RedisCluster::RedisDiscoverySession::onResponse(
     return;
   }
 
-  auto slots = std::make_unique<std::vector<ClusterSlot>>();
-
+  auto cluster_slots = std::make_unique<std::vector<ClusterSlot>>();
+
+  // CLUSTER SLOTS represents nested array of redis instances, like this:
+  //
+  // 1) 1) (integer) 0                <-- start slot range
+  //    2) (integer) 5460             <-- end slot range
+  //
+  //    3) 1) "127.0.0.1"                                   |
+  //       2) (integer) 30001                               <-  master for slot as IP(HOST) / Port /
+  //       ID 3) "09dbe9720cda62f7865eabc5fd8857c5d2678366"    |
+  //
+  //    4) 1) "127.0.0.1"                                   |
+  //       2) (integer) 30004                               <- replicas in the same format as master
+  //       3) "821d8ca00d7ccf931ed3ffc7e3db0599d2271abf"    |
```

Comment:
Should this also say `ID 3)` like the entry above?

Reply:
Ah, looks like the `ID` is extra. Removed in the upper entry.